railgun/cmd/rootcmd/rootcmd.go

```diff
@@ -0,0 +1,29 @@
+package rootcmd
+
+import (
+	"context"
+	"flag"
+
+	"github.com/kong/kubernetes-ingress-controller/railgun/manager"
+	"github.com/spf13/cobra"
+)
+
+var config manager.Config
+
+func init() {
+	fs := flag.NewFlagSet("", flag.ExitOnError)
+	manager.RegisterFlags(&config, fs)
+	rootCmd.Flags().AddGoFlagSet(fs)
+}
+
+var rootCmd = &cobra.Command{
```

Comment:
I can understand some of the potential upsides of using cobra, but using it does break our conformity with the Kubebuilder SDK a bit. I guess I'm OK with this but let's make sure we test all the primary functionality of `kubebuilder` (e.g. create API, controller, e.t.c.) before we merge.

Reply:
We talked about this in Zoom this morning, given time constraints and a desire to focus on testing concerns for KIC 2.0, we're going to opt to intentionally sever controller management with kubebuilder for 2.0, instead using it only for API and config management for this release. We can re-connect the controller management later when we make more progress on splitting the ctrl monolith into separate microservices in later iterations.